### ISSUE TITLE
batch/v1beta1 CronJob is deprecated ... use batch/v1 CronJob

### ISSUE THREAD

**Issue by kingdonb** [fluxcd/flux#3564]
### Describe the bug

```
warnings.go:70] batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob
```

Related to this comment in #3558:

* https://github.com/fluxcd/flux/pull/3558#issuecomment-956543000

We actually call this API by name; when/if Flux v1 should upgrade the K8s client-go again, it will be a good time to address this by moving on. The CronJob batch/v1 API has only been available since 1.21 though, so it may need to wait a while longer before it can be addressed here.

This issue may also never need to be addressed, depending on future developments in the [Migration Timetable](https://fluxcd.io/docs/migration/timetable/).

(Please upgrade to Flux v2 ASAP!)

### Steps to reproduce

Simply run Flux at v1.24.1, this message comes up repeatedly.

It's not nearly as distressing as the list of warnings addressed in #3558 but it does repeat (every automation interval?)

### Expected behavior

No deprecation warnings in the normal course of Flux's operation.

### Kubernetes version / Distro / Cloud provider

kubeadm v1.21.6 on bare metal

### Flux version

Flux v1.24.1+

### Git provider

N/A

### Container Registry provider

N/A

### Maintenance Acknowledgement

- [X] I am aware of Flux v1's maintenance status

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kingdonb** [comment]
Kubernetes 1.24 is due out tomorrow (!)

I think we said we'd target this upgrade to come out with the same timing as about Kubernetes 1.25, so later this year.

Will continue to keep an eye on this and test against the latest releases as they come out.

**Comment by pjbgf** [comment]
This project is in Migration and security support only, so unfortunately this issue won't be fixed. We recommend users to migrate to Flux 2 at their earliest convenience.

More information about the Flux 2 transition timetable can be found at: https://fluxcd.io/docs/migration/timetable/.

**Comment by kingdonb** [comment]
This is the issue to fix, in order to provide support for K8s 1.25 and up. We are probably throwing errors on this version now.

**Comment by kingdonb** [comment]
I will close this issue, as we will be archiving the fluxcd/flux repository soon:

* #3645
